### PR TITLE
Make Jaeger Exporter class and constructor public

### PR DESCRIPTION
**Pull request by vishweshbankwar**
Fixes #.

## Changes

Please provide a brief description of the changes here.
Making Jaeger Exporter constructor and class public: This is needed in order to allow use of [Filtering Processor](https://github.com/open-telemetry/opentelemetry-dotnet/tree/master/docs/trace/extending-the-sdk#filtering-processor)

This change is a part of following issue:
[1612](https://github.com/open-telemetry/opentelemetry-dotnet/issues/1612)

For significant contributions please make sure you have completed the following items:

* [x] `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

**Review by eddynaka — APPROVED** [review]

File: src/OpenTelemetry.Exporter.Jaeger/ApacheThrift/Transport/TTransport.cs [hunk 1]
```diff
@@ -22,7 +22,7 @@
 namespace Thrift.Transport
 {
     // ReSharper disable once InconsistentNaming
-    internal abstract class TTransport : IDisposable
+    public abstract class TTransport : IDisposable
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Exporter.Jaeger/ApacheThrift/Transport/TTransport.cs:25]: I think we should keep this internal. See below.

File: src/OpenTelemetry.Exporter.Jaeger/JaegerExporter.cs [hunk 2]
```diff
@@ -40,7 +40,7 @@ internal class JaegerExporter : BaseExporter<Activity>
         private int batchByteSize;
         private bool disposedValue; // To detect redundant dispose calls
 
-        internal JaegerExporter(JaegerExporterOptions options, TTransport clientTransport = null)
+        public JaegerExporter(JaegerExporterOptions options, TTransport clientTransport = null)
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Exporter.Jaeger/JaegerExporter.cs:43]: I think the `TTransport` is used for test/benchmarks. Could we do...

```csharp
public JaegerExporter(JaegerExporterOptions options) : this(options, null) { }
internal JaegerExporter(JaegerExporterOptions options, TTransport clientTransport = null) { ... }
```

That _should_ allows us to keep the Thrift stuff internal.

Comment (vishweshbankwar) [reply src/OpenTelemetry.Exporter.Jaeger/JaegerExporter.cs]: @CodeBlanch - Thanks for the feedback. I have updated the PR - Please review.

**Review by CodeBlanch — APPROVED** [review]
LGTM

**Review by cijothomas — APPROVED** [review]

**Pull request merged**